Broker cannot start with S3 accessKey and secretKey not supplied

**Description**
We are configuring brokers with the S3 backup properties as follows:
- ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLEBACKUP: true
- ZEEBE_BROKER_DATA_BACKUP_STORE: S3
- ZEEBE_BROKER_DATA_BACKUP_S3_BUCKETNAME: <s3_bucket_name>
- ZEEBE_BROKER_DATA_BACKUP_S3_REGION: us-east-1
- ZEEBE_BROKER_DATA_BACKUP_S3_ENDPOINT: "https://s3.us-east-1.amazonaws.com"

We don't supply credentials through environment variables as they can be pulled from aws credentials provider chain. These properties worked on old version 8.1.9 but we got errors when switching to latest version 8.2.1 and broker cannot start due to the error:

> 
> 2023-04-14T11:21:51.313-04:00 | org.springframework.boot.SpringApplication - Application run failed
> 2023-04-14T11:21:51.313-04:00 | java.lang.IllegalStateException: Failed to execute CommandLineRunner
> 2023-04-14T11:21:51.313-04:00 | at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:772) [spring-boot-3.0.5.jar:3.0.5]
> 2023-04-14T11:21:51.313-04:00 | at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:753) [spring-boot-3.0.5.jar:3.0.5]
> 2023-04-14T11:21:51.313-04:00 | at org.springframework.boot.SpringApplication.run(SpringApplication.java:317) [spring-boot-3.0.5.jar:3.0.5]
> 2023-04-14T11:21:51.313-04:00 | at io.camunda.zeebe.broker.StandaloneBroker.main(StandaloneBroker.java:82) [camunda-zeebe-8.2.1.jar:8.2.1]
> 2023-04-14T11:21:51.313-04:00 | Caused by: io.camunda.zeebe.broker.system.InvalidConfigurationException: Failed configuring backup store S3
> 2023-04-14T11:21:51.313-04:00 | at io.camunda.zeebe.broker.system.SystemContext.validateBackupCfg(SystemContext.java:132) ~[zeebe-broker-8.2.1.jar:8.2.1]
> 2023-04-14T11:21:51.313-04:00 | at io.camunda.zeebe.broker.system.SystemContext.validateDataConfig(SystemContext.java:116) ~[zeebe-broker-8.2.1.jar:8.2.1]
> 2023-04-14T11:21:51.313-04:00 | at io.camunda.zeebe.broker.system.SystemContext.validateConfiguration(SystemContext.java:71) ~[zeebe-broker-8.2.1.jar:8.2.1]
> 2023-04-14T11:21:51.313-04:00 | at io.camunda.zeebe.broker.system.SystemContext.initSystemContext(SystemContext.java:60) ~[zeebe-broker-8.2.1.jar:8.2.1]
> 2023-04-14T11:21:51.313-04:00 | at io.camunda.zeebe.broker.system.SystemContext.<init>(SystemContext.java:56) ~[zeebe-broker-8.2.1.jar:8.2.1]
> 2023-04-14T11:21:51.313-04:00 | at io.camunda.zeebe.broker.StandaloneBroker.run(StandaloneBroker.java:87) ~[camunda-zeebe-8.2.1.jar:8.2.1]
> 2023-04-14T11:21:51.313-04:00 | at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:769) ~[spring-boot-3.0.5.jar:3.0.5]
> 2023-04-14T11:21:51.313-04:00 | ... 3 more
> 2023-04-14T11:21:51.313-04:00 | Caused by: java.lang.NullPointerException: Access key ID cannot be blank.

It looks like io.camunda.zeebe.backup.s3.S3BackupStore.buildClient still calls AwsBasicCredentials.create(credentials.accessKey(), credentials.secretKey() even we don't pass in any credentials. It might be related to this commit: https://github.com/camunda/zeebe/commit/dfd3b9e1034365b3fc1859d5e35353826e1222b3#diff-1b91d4d4e6875c0169ea56f9c35382f5f194a1af782db4879a26971fe32e6daeL72-L74
We also noticed that none of your unit or integration tests are checking for cases where ACCESS_KEY, SECRET_KEY are not passed. 

Could you please look into this issue? Thank you!

--- a/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/backup/S3BackupStoreConfig.java
+++ b/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/backup/S3BackupStoreConfig.java
@@ -8,6 +8,7 @@
 package io.camunda.zeebe.broker.system.configuration.backup;
 
 import io.camunda.zeebe.backup.s3.S3BackupConfig;
+import io.camunda.zeebe.backup.s3.S3BackupConfig.Builder;
 import io.camunda.zeebe.broker.system.configuration.ConfigurationEntry;
 import java.time.Duration;
 import java.util.Objects;
@@ -102,16 +103,19 @@ public class S3BackupStoreConfig implements ConfigurationEntry {
   }
 
   public static S3BackupConfig toStoreConfig(S3BackupStoreConfig config) {
-    return new S3BackupConfig.Builder()
-        .withBucketName(config.getBucketName())
-        .withEndpoint(config.getEndpoint())
-        .withRegion(config.getRegion())
-        .withCredentials(config.getAccessKey(), config.getSecretKey())
-        .withApiCallTimeout(config.getApiCallTimeout())
-        .forcePathStyleAccess(config.isForcePathStyleAccess())
-        .withCompressionAlgorithm(config.getCompression())
-        .withBasePath(config.getBasePath())
-        .build();
+    final var builder =
+        new Builder()
+            .withBucketName(config.getBucketName())
+            .withEndpoint(config.getEndpoint())
+            .withRegion(config.getRegion())
+            .withApiCallTimeout(config.getApiCallTimeout())
+            .forcePathStyleAccess(config.isForcePathStyleAccess())
+            .withCompressionAlgorithm(config.getCompression())
+            .withBasePath(config.getBasePath());
+    if (config.getAccessKey() != null && config.getSecretKey() != null) {
+      builder.withCredentials(config.getAccessKey(), config.getSecretKey());
+    }
+    return builder.build();
   }
 
   @Override

--- a/qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/S3BackupAuthenticationIT.java
+++ b/qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/S3BackupAuthenticationIT.java
@@ -1,0 +1,57 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.it.backup;
+
+import io.camunda.zeebe.qa.util.testcontainers.MinioContainer;
+import io.camunda.zeebe.qa.util.testcontainers.ZeebeTestContainerDefaults;
+import io.camunda.zeebe.test.util.junit.RegressionTest;
+import io.zeebe.containers.ZeebeContainer;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Network;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+final class S3BackupAuthenticationIT {
+  private static final Network NETWORK = Network.newNetwork();
+  private static final String BUCKET_NAME = RandomStringUtils.randomAlphabetic(10).toLowerCase();
+
+  @Container
+  private static final MinioContainer MINIO =
+      new MinioContainer().withNetwork(NETWORK).withDomain("minio.local", BUCKET_NAME);
+
+  @Test
+  @RegressionTest("https://github.com/camunda/zeebe/issues/12433")
+  void shouldConnectWithoutConfiguredCredentials() {
+    // given
+    final var zeebe =
+        new ZeebeContainer(ZeebeTestContainerDefaults.defaultTestImage())
+            .withNetwork(NETWORK)
+            .dependsOn(MINIO)
+            .withoutTopologyCheck()
+            .withEnv("MANAGEMENT_ENDPOINTS_WEB_EXPOSURE_INCLUDE", "*")
+            .withEnv("ZEEBE_BROKER_DATA_BACKUP_STORE", "S3")
+            .withEnv("ZEEBE_BROKER_DATA_BACKUP_S3_BUCKETNAME", BUCKET_NAME)
+            .withEnv("ZEEBE_BROKER_DATA_BACKUP_S3_ENDPOINT", MINIO.internalEndpoint())
+            .withEnv("ZEEBE_BROKER_DATA_BACKUP_S3_REGION", MINIO.region())
+            // Set env variables discovered by the AWS SDK, not Zeebe
+            .withEnv("AWS_ACCESS_KEY_ID", MINIO.accessKey())
+            .withEnv("AWS_SECRET_ACCESS_KEY", MINIO.secretKey());
+
+    // when
+    zeebe.start();
+
+    // then
+    Assertions.assertThat(zeebe.isStarted()).isTrue();
+
+    // cleanup
+    zeebe.close();
+  }
+}

